modules/xfem/src/base/XFEM.C

```diff
@@ -2042,3 +2125,84 @@ XFEM::getNodeSolutionDofs(const Node * node, SystemBase & sys) const
   }
   return solution_dofs;
 }
+
+const GeometricCutUserObject *
+XFEM::getGeometricCutForElem(const Elem * elem) const
+{
+  for (auto gcmit : _geom_marker_id_elems)
+  {
+    std::set<unsigned int> elems = gcmit.second;
+    if (elems.find(elem->id()) != elems.end())
+      return _geometric_cuts[gcmit.first];
+  }
+  return nullptr;
+}
+
+void
+XFEM::storeMaterialPropertiesForElements(const Elem * parent_elem,
+                                         const Elem * elem1,
+                                         const Elem * elem2)
+{
+  HashMap<unsigned int, MaterialProperties> elem1_props, elem1_props_old, elem1_props_older,
+      elem2_props, elem2_props_old, elem2_props_older;
+  elem1_props = (*_material_data)[0]->getMaterialPropertyStorage().props().at(elem1);
+  elem2_props = (*_material_data)[0]->getMaterialPropertyStorage().props().at(elem2);
+  elem1_props_old = (*_material_data)[0]->getMaterialPropertyStorage().propsOld().at(elem1);
+  elem2_props_old = (*_material_data)[0]->getMaterialPropertyStorage().propsOld().at(elem2);
+  if ((*_material_data)[0]->getMaterialPropertyStorage().hasOlderProperties())
+  {
+    elem1_props_older = (*_material_data)[0]->getMaterialPropertyStorage().propsOlder().at(elem1);
+    elem2_props_older = (*_material_data)[0]->getMaterialPropertyStorage().propsOlder().at(elem2);
+  }
+  auto props_pair = std::make_pair(elem1_props, elem2_props);
+  auto props_old_pair = std::make_pair(elem1_props_old, elem2_props_old);
+  auto props_older_pair = std::make_pair(elem1_props_older, elem2_props_older);
+  auto props_used = std::make_pair(false, false);
+  _healed_material_properties.emplace(parent_elem, props_pair);
+  _healed_material_properties_old.emplace(parent_elem, props_old_pair);
+  _healed_material_properties_older.emplace(parent_elem, props_older_pair);
+  _healed_material_properties_used.emplace(parent_elem, props_used);
+}
+
+void
+XFEM::setMaterialPropertiesForElement(
+    const Elem * parent_elem,
+    const Elem * cut_elem,
+    const HashMap<unsigned int, MaterialProperties> & props,
+    const HashMap<unsigned int, MaterialProperties> & props_old,
+    const HashMap<unsigned int, MaterialProperties> & props_older) const
+{
+  (*_material_data)[0]->copy(*cut_elem, props, props_old, props_older, 0);
+  for (unsigned int side = 0; side < parent_elem->n_sides(); ++side)
+  {
+    std::vector<boundary_id_type> parent_boundary_ids;
+    _mesh->boundary_info->boundary_ids(parent_elem, side, parent_boundary_ids);
+    for (auto bdid : parent_boundary_ids)
+    {
+      if (_fe_problem->needBoundaryMaterialOnSide(bdid, 0))
+        (*_bnd_material_data)[0]->copy(*cut_elem, props, props_old, props_older, side);
+    }
+  }
+}
+
+bool
+XFEM::getElemConsistentSideBoolean(const Elem * parent_elem,
```

Comment:
I think we still want to rename this method, right?

Reply:
`getElementSideOfInterface`?

Reply:
`getElementSideRelativeToInterface`?

Reply:
I thought we are going to rename this method, along with the data structure and all the logics, in the next PR. Because later on we want to properly manage a vector of cut side IDs.

Anyways, I renamed it just for now.